scheduler/src/cook/mesos/util.clj

```diff
@@ -91,22 +92,35 @@
 (defn entity->map
   "Takes a datomic entity and converts it along with any nested entities
   into clojure maps"
-  ([entity & {:keys [db skip-keys]
-              :or {skip-keys #{}}}]
-   (let [entity' (if db (d/entity db (:db/id entity))
-                     entity)]
-     (deep-transduce-kv (filter (fn [[k _]] (not (contains? skip-keys k)))) entity'))))
-
-(defn job-ent->map
-  "Convert a job entity to a map. This also loads the associated group and converts it to a map."
-  ([job db]
-   (job-ent->map (d/entity db (:db/id job))))
-  ([job]
-   (-> (entity->map job)
-       ; Load the group related to the job (if any) but without loading the job entities
-       ; inside the group entity. If someone needs to access them, it's better to query
-       ; at that point instead of relying on the cached data.
-       (assoc :group/_job (entity->map (:group/_job job) :skip-keys #{:group/job})))))
+  ([entity db]
+   (entity->map (d/entity db (:db/id entity))))
+  ([entity]
+   (deep-transduce-kv (map identity) entity)))
+
+; During testing, we found that calling (keys) on a group entity with a large number of jobs was
+; quite slow. As an optimization, load the group keys ahead of time from the schema.
+(let [group-keys (->> schema/schema-attributes
+                      (map :db/ident)
+                      (filter #(= "group" (namespace %)))
+                      ; Do not load the nested job entities for a group. These should be queried on-demand
+                      ; as required.
+                      (filter #(not= :group/job %)))]
+  (defn job-ent->map
+    "Convert a job entity to a map. This also loads the associated group and converts it to a map."
+    ([job db]
+     (job-ent->map (d/entity db (:db/id job))))
+    ([job]
+     (let [group-ent (first (:group/_job job))
+           job (entity->map job)
+           group (when group-ent
+                   (->> group-keys
+                        (map (fn [k] [k (entity->map (get group-ent k))]))
```

Comment:
nitpick: `pc/map-from-keys` might be a better fit here

Reply:
IIRC it was faster this way (with the nil filtering before making the map) but I don't think it was really compelling either way. I kind of like this style but I am also open to changing it.

Reply:
I prefer map-from-keys too. I think it is also ok to leave the nil values

Reply:
Done

Reply:
I don't see the change to `map-from-keys`. Did you forget to push that commit?